### PR TITLE
Support for proxies 

### PR DESCRIPTION
**Pull request by mkcello96**
A working version of the Transifex client with added support for proxies. To use, set the http_proxy and https_proxy environment variables. @kouk

### CODE REVIEW

**Comment by kouk** [comment]
Thanks for this, we'll look into merging and releasing ASAP.

File: txclib/utils.py [hunk 1]
```diff
@@ -105,27 +111,57 @@ def determine_charset(response):
...
+    else:
+        managers["http"] = urllib3.PoolManager(num_pools=num_pools)
+
+    if "https_proxy" in os.environ:
+        proxy_url = os.environ["https_proxy"]
+        managers["https"] = urllib3.ProxyManager(
+            proxy_url=proxy_url,
+            proxy_headers={"User-Agent": user_agent_identifier()},
+            num_pools=num_pools,
+            cert_reqs=ssl.CERT_REQUIRED,
```
Comment (tabac) [inline txclib/utils.py]: @mkcello96 Why are you using  `ssl.CERT_REQUIRED` here above we've imported and used `txclib.config.CERT_REQUIRED`

File: txclib/utils.py [hunk 2]
```diff
@@ -105,27 +111,57 @@ def determine_charset(response):
 
 def make_request(method, host, url, username, password, fields=None,
                  skip_decode=False):
-    charset = None
-    if host.lower().startswith('https://'):
-        connection = urllib3.connection_from_url(
-            host,
-            cert_reqs=CERT_REQUIRED,
+    # Initialize http and https pool managers
+    num_pools = 1
```
Comment (tabac) [inline txclib/utils.py]: That would be great if it was a setting.

File: txclib/utils.py [hunk 3]
```diff
@@ -105,27 +111,57 @@ def determine_charset(response):
...
+        )
+    
+    charset = None
     headers = urllib3.util.make_headers(
         basic_auth='{0}:{1}'.format(username, password),
         accept_encoding=True,
         user_agent=user_agent_identifier(),
         keep_alive=True
     )
+    request = Request(host + url, None, headers)
```
Comment (tabac) [inline txclib/utils.py]: Why are you using `Request` here and not just pass `headers` and `url` to `manager.request(...)` directly?

**Comment by tabac** [comment]
@mkcello96 Thanks a lot for your PR! I have a made a few small recommendations. Also some of your changes broke a few tests, which can be easily fixed since it's just a `mock` that has to change. 

Let me know what you think.
Tasos.

File: txclib/utils.py (hunk 1 above)
Comment (mkcello96) [inline txclib/utils.py]: Thanks- didn't see that

File: txclib/utils.py (hunk 3 above)
Comment (mkcello96) [inline txclib/utils.py]: Got this one fixed too. I got a lot of the code for this from the fix-proxies branch, which was old and didn't work, so I'm guessing that was part of it.

File: txclib/utils.py (hunk 2 above)
Comment (mkcello96) [inline txclib/utils.py]: As for this one, I believe my code still provides the same functionally with the https:// hostnames (in managers["https"]). Were you suggesting I add additional support?

**Comment by mkcello96** [comment]
Thanks for your helpful suggestions - I pushed the fixes for them. As for fixing the broken tests, I'm having a little bit of trouble running the test suite due an old version of setuptools installed on my machine. I'll keep looking at that and keep you updated. @kouk @tabac

**Comment by dimoschi** [comment]
@mkcello96 The reason CircleCI tests are failing is that it doesn't export Project environmental variables, so it cannot send data to codecov. I've pushed PR #132 which solves that, but I need someone else to merge it.

**Comment by mkcello96** [comment]
@dimoschi thank you for your fix! I've merged it into this pull request and all tests are passing. @kouk @tabac - hoping to push this to master ASAP.

**Comment by kouk** [comment]
@dimoschi @mkcello96 thanks for your work. Green light from me.

**Comment by mkcello96** [comment]
Thanks- have a specific date when this will be released next week? @kouk

**Comment by kouk** [comment]
@mkcello96 no specific date I'm afraid but I'm pushing for Monday.

**Comment by mkcello96** [comment]
Any progress on this?  @kouk

**Comment by kouk** [comment]
@mkcello96 version 0.12.2 is released!

**Comment by mkcello96** [comment]
Thanks!! It's working great

**Pull request merged**